src/sentry/overwatch_webhooks/webhook_forwarder.py

```diff
@@ -50,17 +59,79 @@ class OverwatchGithubWebhookForwarder:
     def __init__(self, integration: Integration):
         self.integration = integration
 
-    def should_forward_to_overwatch(self, headers: Mapping[str, str]) -> bool:
+    def _is_forwardable_event_type(self, headers: Mapping[str, str]) -> bool:
+        """Check if the event type is one we should forward to Overwatch."""
         event_type = headers.get(GITHUB_WEBHOOK_TYPE_HEADER_KEY)
         verbose_log(
-            "overwatch.debug.should_forward_to_overwatch.checked",
+            "overwatch.debug.is_forwardable_event_type.checked",
             extra={
                 "event_type": event_type,
                 "should_forward": event_type in GITHUB_EVENTS_TO_FORWARD_OVERWATCH,
             },
         )
         return event_type in GITHUB_EVENTS_TO_FORWARD_OVERWATCH
 
+    def _get_pull_request_author_id(self, event: Mapping[str, Any]) -> str | None:
+        """
+        Extract the pull request author's external identifier from the webhook event.
+
+        We specifically want the PR author (not the commenter/reviewer) because
+        that's whose seat we need to check for billing purposes.
+        """
+        if "pull_request" not in event:
+            return None
+
+        user = event["pull_request"].get("user", {})
+        if user_id := user.get("id"):
+            return str(user_id)
+
+        return None
+
+    def _filter_eligible_org_summaries(
+        self,
+        org_summaries: list[OrganizationSummary],
+        event: Mapping[str, Any],
+    ) -> list[OrganizationSummary]:
+        """Filter org_summaries to only include orgs eligible for Overwatch forwarding."""
+        repo_external_id = event.get("repository", {}).get("id")
+        external_identifier = self._get_pull_request_author_id(event)
+
+        if not repo_external_id or not external_identifier:
+            return []
+
+        eligible_summaries: list[OrganizationSummary] = []
+        org_ids = [summary.id for summary in org_summaries]
+
+        # Batch fetch organizations (needed for feature flag check)
+        orgs_by_id = {org.id: org for org in Organization.objects.filter(id__in=org_ids)}
+
+        # Batch fetch repository IDs (we only need id and organization_id)
+        repos_by_org: dict[int, int] = {
+            repo["organization_id"]: repo["id"]
+            for repo in Repository.objects.filter(
+                organization_id__in=org_ids,
+                external_id=str(repo_external_id),
+                provider=f"integrations:{self.integration.provider}",
+            ).values("id", "organization_id")
+        }
```

Comment:
`Repository` is a region-silo model, so we won't be able to query it here. If we don't have tests failing with silo errors for this, we probably need to add them.